extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -401,12 +401,22 @@ object PrivilegesBuilder {
         buildQuery(getQuery, inputObjs)
 
       case "InsertIntoDataSourceDirCommand" |
-          "SaveIntoDataSourceCommand" |
-          "InsertIntoHadoopFsRelationCommand" |
           "InsertIntoHiveDirCommand" =>
+        buildQuery(getQuery, inputObjs)
+
+      case "SaveIntoDataSourceCommand" =>
         // TODO: Should get the table via datasource options?
         buildQuery(getQuery, inputObjs)
 
+      case "InsertIntoHadoopFsRelationCommand" =>
+        getPlanField[Option[CatalogTable]]("catalogTable").map(opt => {
```

Comment:
Can you verify if there are any compilation errors for other spark versions, and attach the result in the PR description

Reply:
Other versions means spark3.1/spark3.2/spark3.3?

Reply:
yes

Reply:
and maybe 2.4 and 3.0

Reply:
I tested different Spark version(2.4.0、3.0.0、3.1.3、3.2.2、3.3.0) with a simple script, the conclusion is that:

- all versions compile normally except 2.4;
- and the cause of 2.4 compile failure has nothing to do with this PR(fear is premature failure, leafing through the Spark code to find should be irrelevant, see: [spark2.4 code](https://github.com/apache/spark/blob/4be566062defa249435c4d72eb106fe7b933e023/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala#L57)

**script:**
```
#!/bin/bsh

versions=("2.4.0" "3.0.0" "3.1.3" "3.2.2" "3.3.0")

last=3.2.2
for e in ${versions[@]}; do
	#statements
	echo "Building Spark version $e"

	sed -i "" "s/\<spark\.version\>$last\<\/spark\.version\>/\<spark\.version\>$e\<\/spark\.version\>/g" /Users/yikaifei/code/kyuubi/pom.xml
	last=$e

	./build/mvn clean package -DskipTests -T 4 -pl 'extensions/spark/kyuubi-spark-authz' -am -o
done


sed -i "" "s/\<spark\.version\>3.3.0\<\/spark\.version\>/\<spark\.version\>3.2.2\<\/spark\.version\>/g" /Users/yikaifei/code/kyuubi/pom.xml
```
**result:**
```
yikaifei@B000000335049P kyuubi % sh Muti-Spark-Auth.sh
Building Spark version 2.4.0
Using `mvn` from path: /Users/yikaifei/code/kyuubi/build/apache-maven-3.8.4/bin/mvn
[ERROR] /Users/yikaifei/code/kyuubi/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleAuthorization.scala:25: object TreeNodeTag is not a member of package org.apache.spark.sql.catalyst.trees
[ERROR] /Users/yikaifei/code/kyuubi/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleAuthorization.scala:44: not found: value TreeNodeTag
[ERROR] /Users/yikaifei/code/kyuubi/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleAuthorization.scala:34: value getTagValue is not a member of org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
[ERROR] /Users/yikaifei/code/kyuubi/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleAuthorization.scala:36: value setTagValue is not a member of org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
[ERROR] four errors found
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile (scala-compile-first) on project kyuubi-spark-authz_2.12: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile failed: Compilation failed: InterfaceCompileFailed -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :kyuubi-spark-authz_2.12
Building Spark version 3.0.0
Using `mvn` from path: /Users/yikaifei/code/kyuubi/build/apache-maven-3.8.4/bin/mvn
Building Spark version 3.1.3
Using `mvn` from path: /Users/yikaifei/code/kyuubi/build/apache-maven-3.8.4/bin/mvn
Building Spark version 3.2.2
Using `mvn` from path: /Users/yikaifei/code/kyuubi/build/apache-maven-3.8.4/bin/mvn
Building Spark version 3.3.0
Using `mvn` from path: /Users/yikaifei/code/kyuubi/build/apache-maven-3.8.4/bin/mvn
```



BTW:  there seems to be a problem unrelated to this PR when `./build/mvn clean package -DskipTests -T 4 -Pspark-3.1,spark-3.2,spark-3.3`
<img width="1436" alt="image" src="https://user-images.githubusercontent.com/51110188/185613310-548c16b3-5975-4f94-8456-f4e80346fee3.png">

Reply:
> BTW: there seems to be a problem unrelated to this PR when `./build/mvn clean package -DskipTests -T 4 -Pspark-3.1,spark-3.2,spark-3.3`

If it is confirmed to be a problem, i will fix two points:
1. Why didn't GA find this
2. fix these errors

Reply:
> `-Pspark-3.1,spark-3.2,spark-3.3`

we don't support it because the shared common module compiled only once and it maybe not compatiable w/ echo version.

Reply:
BTW, the maven properties can override by `-Dk=v`, e.g. 
```
build/mvn clean install -DskipTests -Pspark-3.1 -Dspark.version=3.1.1
```

Reply:
oh, yea ☺, thanks~